core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -1148,8 +1137,7 @@ public static IndexMetaData fromXContent(XContentParser parser) throws IOExcepti
                         assert Version.CURRENT.major <= 5;
                         parser.skipChildren();
                     } else {
-                        // check if its a custom index metadata
-                        Custom proto = lookupPrototype(currentFieldName);
+                        Custom proto = registry.getIndexMetadataPrototype(currentFieldName);
                         if (proto == null) {
                             //TODO warn
```

Comment:
this is existing behavior but it's bad... forgetting to install a plugin immediately implies the deletion of it's state (and solving it will be tricky). Can we add this warn log and open an issue to deal with it? (maybe the plugin manager to should remove custom meta when uninstalling. not sure)